resources/prosody-plugins/mod_speakerstats_component.lua

```diff
@@ -0,0 +1,189 @@
+local get_room_from_jid = module:require "util".get_room_from_jid;
+local jid_resource = require "util.jid".resource;
+local ext_events = module:require "ext_events"
+local st = require "util.stanza";
+local socket = require "socket";
+local json = require "util.json";
+
+local muc_component_host = module:get_option_string("muc_component");
+if muc_component_host == nil then
+    log("error", "No muc_component specified. No muc to operate on!");
+    return;
+end
+local muc_module = module:context("conference."..muc_component_host);
+if muc_module == nil then
+    log("error", "No such muc found, check muc_component config.");
+    return;
+end
+
+log("debug", "Starting speakerstats for %s", muc_component_host);
+
+-- receives messages from client currently connected to the room
+-- clients indicates their own dominant speaker events
+function on_message(event)
+    -- Check the type of the incoming stanza to avoid loops:
+    if event.stanza.attr.type == "error" then
+        return; -- We do not want to reply to these, so leave.
+    end
+
+    local speakerStats
+        = event.stanza:get_child('speakerstats', 'http://jitsi.org/jitmeet');
+    if speakerStats then
+        local roomAddress = speakerStats.attr.room;
+        local room = get_room_from_jid(roomAddress);
+
+        if not room then
+            log("warn", "No room found %s", roomAddress);
+            return false;
+        end
+
+        local roomSpeakerStats = room.speakerStats;
+        local from = event.stanza.attr.from;
+
+        local occupant = room:get_occupant_by_real_jid(from);
+        if not occupant then
+            log("warn", "No occupant %s found for %s", from, roomAddress);
+            return false;
+        end
+
+        local newDominantSpeaker = roomSpeakerStats[occupant.jid];
+        local oldDominantSpeakerId = roomSpeakerStats['dominantSpeakerId'];
+
+        if oldDominantSpeakerId then
+            roomSpeakerStats[oldDominantSpeakerId]:setIsDominantSpeaker(false);
+        end
+
+        if newDominantSpeaker then
+            newDominantSpeaker:setIsDominantSpeaker(true);
+        end
+
+        room.speakerStats['dominantSpeakerId'] = occupant.jid;
+    end
+
+    return true
+end
+
+--- Start SpeakerStats implementation
+local SpeakerStats = {};
+SpeakerStats.__index = SpeakerStats;
+
+function new_SpeakerStats(nick)
+    return setmetatable({
+        totalDominantSpeakerTime = 0;
+        _dominantSpeakerStart = nil;
+        _isDominantSpeaker = false;
+        nick = nick;
+        displayName = nil;
+    }, SpeakerStats);
+end
+
+-- Changes the dominantSpeaker data for current occupant
+-- saves start time if it is new dominat speaker
+-- or calculates and accumulates time of speaking
+function SpeakerStats:setIsDominantSpeaker(isNowDominantSpeaker)
```

Comment:
`setIsDominantSpeaker` sounds a little strange. How about `setDominantSpeaker`?

Reply:
Well, I copied the logic from the js, and would prefer to keep the names the same: https://github.com/jitsi/lib-jitsi-meet/blob/e709caaa7c70eb3e340e1b65348ae1040c32d358/modules/statistics/SpeakerStats.js#L83

Reply:
I changed it in both jitsi-meet and lib-jitsi-meet.